### PR TITLE
Implemented CommonTimeBlock

### LINKED ISSUE #13: [SubIssue 1.5] TimeBlockclasses

### PR DESCRIPTION
**Pull request by MMachadoUofT**
I definitely see why TDD is so highly regarded... lots of tiny bugs I wouldn't have been able to find otherwise. I made sure to include thorough tests to assure that this suite was as penetrative as possible. The rest of the remaining classes are very simple, as they are basically just containers for the ones I've created thus far, and any of their functionality will primarily involve calling their sub-entities' methods.

### CODE REVIEW

**Review by Amir1441 — DISMISSED** [review]

**Review by Eggathin — DISMISSED** [review]

File: src/main/java/entity_layer/CommonTimeBlock.java [hunk 1]
```diff
@@ -11,6 +12,26 @@
...
+    private LocalDateTime startTime;
+    private LocalDateTime endTime;
+
+    /* ************ *\
+    *  Constructors  *
+    \* ************ */
+    public CommonTimeBlock(LocalDateTime startTime, LocalDateTime endTime) {
+        this.startTime = startTime;
+        this.endTime = endTime;
+        // I'll have to include a check somewhere that doesn't allow start time to be before
```
Comment (dadler83) [inline src/main/java/entity_layer/CommonTimeBlock.java]: nonsensical comment

File: src/main/java/entity_layer/TimeBlock.java [hunk 2]
```diff
@@ -27,7 +27,8 @@ public interface TimeBlock {
     int hashCode();
```
Comment (dadler83) [inline src/main/java/entity_layer/TimeBlock.java]: I don't think you need hashCode here because it is overridden from the object class

**Review by dadler83 — CHANGES_REQUESTED** [review]
If the tests all pass, it looks pretty good. I left some minor comments about cleaning up the code though, nothing functionality wise to comment on -- it looks good..

File: src/main/java/entity_layer/CommonTimeBlock.java (hunk 1 above)
Comment (dadler83) [reply src/main/java/entity_layer/CommonTimeBlock.java]: If you are going to implement the check do so, if not, let a higher level object implement it

**Review by dadler83 — APPROVED** [review]
These entities look good for now; we'll start using them for the use cases, and clean up code as needed.

**Review by Amir1441 — APPROVED** [review]
The methods look more and more complete now. Each one is more tuned into the specific role it has to play in the entirety of the project. Looks good. With this, the Time Block usecases can now run

**Review by Eggathin — APPROVED** [review]

**Pull request merged**